neurots/generate/tree.py

```diff
@@ -172,6 +172,20 @@ def add_section(
             children (int): The number of children.
         """
         SGrower = section_growers[self.params["metric"]]
+        context = copy.deepcopy(self.context)
+        if self.context is not None and "constraints" in self.context:  # pragma: no cover
+            constraints = [
+                constraint
+                for constraint in self.context["constraints"]
+                if "section_prob" in constraint
+                and SectionType(self.params["tree_type"]).name
+                in constraint.get("neurite_types", [])
+                and process in constraint.get("processes", ["major", "secondary"])
+            ]
+            if constraints:
+                context["constraints"] = constraints
+            else:
+                del context["constraints"]
```

Comment:
Why do you prefer to delete it instead of keeping an empty list?

Reply:
Ok but I feel it dangerous because the result of `SectionGrower.next_point()` will be different whether `self.context["constraints"]` does not exist or if it exists with an empty list for example. This means that you suppose that `SectionGrower.next_point()` is always called after `TreeGrower.add_section()`, but this is actually not mandatory.
Maybe it would be safer to do something like `if context.get("constraints", [])` in `SectionGrower.next_point()`? And so you can leave an empty list in `TreeGrower.add_section()`.

Reply:
I think it's safer :)

Reply:
done!